### PR TITLE
Blog app - Processing data in models

### PR DESCRIPTION
**Pull request by anthonymr**
## Milestone 2: Processing data in models

In this first PR, I completed the following steps:

- Create model classes for all entities as shown in the [ERD diagram](../images/blog_app_erd.png).
- Set up associations between models.
- Create custom methods as follow:
  -  `Users`
      - A method that returns the 3 most recent posts for a given user.
  -  `Posts`
      - A method that updates the posts counter for a user.
      - A method that returns the 5 most recent comments for a given post.
  -  `Comments`
      - A method that updates the comments counter for a post.
  -  `Likes`
      - A method that updates the likes counter for a post.

ℹ️ To run this project:

- Clone this repository to your local machine:

```sh

  git clone git@github.com:anthonymr/anthony-blog.git
  cd anthony-blog
```
- Install dependencies:
```sh
  bundle install
```
- Recreate Data Base:
```sh
  rake db:schema:load
```
Run `Rails` server:

```  sh
  rails s  
```

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class User < ApplicationRecord
+  def self.last_posts(user)
+    return unless user
+
+    Post.where(author_id: user[:id]).order(created_at: :desc).limit(3)
+  end
```
Comment (mikethreels) [inline app/models/user.rb]: - [x] the user has many relationship with post, like and comment should be set in this class

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+
+  def self.last_comments(post)
+    return unless post
```
Comment (mikethreels) [inline app/models/post.rb]: - [x] the posts has many relationship with comment and like should be set in this class

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Anthony, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- good job setting the belongs_to relationships.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 2 above)
Comment (anthonymr) [reply app/models/post.rb]: Hi @mikethreels, thanks for your review. I'll apply those changes.

**Review by alexander16108 — APPROVED** [review]
Hi @anthonymr , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- good job setting the belongs_to relationships.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**